wallet/internal/db/data_types.go

```diff
@@ -0,0 +1,800 @@
+		Purpose: BIP0044Purpose,
+		Coin:    0,
+	}
+
+	// ScopeAddrMap is a map from the default key scopes to the scope
+	// address schema for each scope type.
+	ScopeAddrMap = map[KeyScope]ScopeAddrSchema{
+		KeyScopeBIP0049Plus: {
+			ExternalAddrType: NestedWitnessPubKey,
+			InternalAddrType: WitnessPubKey,
+		},
+		KeyScopeBIP0084: {
+			ExternalAddrType: WitnessPubKey,
+			InternalAddrType: WitnessPubKey,
+		},
+		KeyScopeBIP0086: {
+			ExternalAddrType: TaprootPubKey,
+			InternalAddrType: TaprootPubKey,
+		},
+		KeyScopeBIP0044: {
+			InternalAddrType: PubKeyHash,
+			ExternalAddrType: PubKeyHash,
+		},
+	}
+)
+
+// Tapscript represents a Taproot script leaf, which includes the script itself
+// and its corresponding control block. This is used for spending Taproot
+// outputs.
+type Tapscript struct {
+	// ControlBlock is the control block for the Taproot script, which is
+	// required to reveal the script path during spending.
+	ControlBlock []byte
+
+	// Script is the actual script code of the Taproot leaf.
+	Script []byte
+}
+
+// --------------------
+// WalletStore Types
+// --------------------
+
+// WalletInfo contains the static properties of a wallet. This struct provides a
+// summary of the wallet's configuration and state.
+type WalletInfo struct {
+	// ID is the unique identifier for the wallet.
+	ID uint64
+
+	// Name is the human-readable name of the wallet.
+	Name string
+
+	// IsImported indicates whether the wallet was created from an existing
+	// seed or was created as a new wallet.
+	IsImported bool
+
+	// IsWatchOnly indicates whether the wallet is in watch-only mode,
+	// meaning it does not have private keys and cannot sign transactions.
+	IsWatchOnly bool
+
+	// Birthday is the timestamp of the wallet's creation, used as a
+	// starting point for rescans.
+	Birthday time.Time
+
+	// BirthdayBlock is the block hash and height from which to start a
+	// rescan.
+	BirthdayBlock Block
+
+	// SyncedTo represents the wallet's current synchronization state with
+	// the blockchain.
+	SyncedTo Block
+}
+
+// Block defines a block's hash, height, and timestamp. This is used to
+// represent a block's identity and position in the blockchain.
+type Block struct {
+	// Hash is the 32-byte hash of the block.
+	Hash chainhash.Hash
+
+	// Height is the height of the block in the blockchain.
+	Height int32
+
+	// Timestamp is the timestamp of the block, which is used for wallet
+	// synchronization and rescan operations.
+	Timestamp time.Time
+}
+
+// CreateWalletParams contains the parameters required to create a new wallet.
+type CreateWalletParams struct {
+	// Name is the name of the new wallet.
+	Name string
+
+	// PublicPassphrase is the passphrase used to encrypt public data.
+	PublicPassphrase []byte
+
+	// PrivatePassphrase is the passphrase used to encrypt private keys.
+	PrivatePassphrase []byte
+
+	// HDSeed is the Hierarchical Deterministic (HD) seed for the new
+	// wallet.
+	HDSeed []byte
```

Comment:
```suggestion
	// HDSeed represents the binary seed used to derive the master private key
    // in an HD wallet as defined by BIP32.
	HDSeed []byte
```
Maybe could clarify a little bit what seed we're talking about.